samples/nrf9160/gnss/src/assistance_minimal.c

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <zephyr.h>
+#include <sys/timeutil.h>
+#include <logging/log.h>
+#include <nrf_modem_at.h>
+#include <nrf_modem_gnss.h>
+
+#include "assistance.h"
+#include "factory_almanac.h"
+#include "mcc_location_table.h"
+
+LOG_MODULE_DECLARE(gnss_sample, CONFIG_GNSS_SAMPLE_LOG_LEVEL);
+
+/* (6.1.1980 UTC - 1.1.1970 UTC) */
+#define GPS_TO_UNIX_UTC_OFFSET_SECONDS	(315964800UL)
+/* UTC/GPS time offset as of 1st of January 2017. */
+#define GPS_TO_UTC_LEAP_SECONDS		(18UL)
+#define SEC_PER_MIN			(60UL)
+#define MIN_PER_HOUR			(60UL)
+#define SEC_PER_HOUR			(MIN_PER_HOUR * SEC_PER_MIN)
+#define HOURS_PER_DAY			(24UL)
+#define SEC_PER_DAY			(HOURS_PER_DAY * SEC_PER_HOUR)
+#define DAYS_PER_WEEK			(7UL)
+#define PLMN_STR_MAX_LEN		8 /* MCC + MNC + quotes */
+
+static void factory_almanac_write(void)
+{
+	if (nrf_modem_at_printf("AT%%XFILEWRITE=1,\"%s\",\"%s\"",
```

Comment:
I think that this causes data to be written to flash every time, so this is not something that we want to do on every boot. There's no way to check from modem side whether the same file already exists, so we should probably store this information on the application side.

For example, we could store the file checksum using the settings subsystem and use that to check whether the same almanac has already been written once.

Reply:
Added implementation which stores the almanac checksum and only writes factory almanac to flash if the checksum doesn't match.